src/openshift/storage.ts

```diff
@@ -10,31 +10,84 @@ import * as validator from 'validator';
-        const component: OpenShiftObject = context;
-        const app: OpenShiftObject = context.getParent();
+        const data: OpenShiftObject = await Storage.getComponentData(context);
+        const component: OpenShiftObject = data;
+        const app: OpenShiftObject = data.getParent();
         const project: OpenShiftObject = app.getParent();
-        const storageName = await vscode.window.showInputBox({prompt: "Specify the storage name", validateInput: (value: string) => {
-            if (validator.isEmpty(value.trim())) {
-                return 'Invalid storage name';
-            }
-        }});
-        if (!storageName) return null;
+        if (data) {
+            const storageName = await vscode.window.showInputBox({prompt: "Specify the storage name", validateInput: (value: string) => {
+                if (validator.isEmpty(value.trim())) {
+                    return 'Invalid storage name';
+                }
+            }});
+            if (!storageName) return null;
 
-        const mountPath = await vscode.window.showInputBox({prompt: "Specify the mount path", validateInput: (value: string) => {
-            if (validator.isEmpty(value.trim())) {
-                return 'Invalid mount path';
-            }
-        }});
-        if (!mountPath) return null;
+            const mountPath = await vscode.window.showInputBox({prompt: "Specify the mount path", validateInput: (value: string) => {
+                if (validator.isEmpty(value.trim())) {
+                    return 'Invalid mount path';
+                }
+            }});
+            if (!mountPath) return null;
+
+            const storageSize = await vscode.window.showQuickPick(['1Gi', '1.5Gi', '2Gi'], {placeHolder: 'Select the storage size'});
+            if (!storageSize) return null;
+
+            return Promise.resolve()
+                .then(() => Storage.odo.execute(Command.createStorage(project.getName(), app.getName(), component.getName(), storageName, mountPath, storageSize)))
+                .then(() => Storage.explorer.refresh())
+                .then(() => `Storage '${storageName}' successfully created for component '${data.getName()}'`)
+                .catch((err) => Promise.reject(`New Storage command failed with error: '${err}'!`));
+
+        }
+    }
+
+    static async getComponentData(component) {
+        let componentName: OpenShiftObject  = component ? component : undefined;
+        let name: OpenShiftObject;
+        if (!componentName) {
+            name = await Storage.getProjectName();
+        }
+        if (componentName) {
+            componentName = component;
+        }
+        return name || componentName ? name || componentName : undefined;
+    }
 
-        const storageSize = await vscode.window.showQuickPick(['1Gi', '1.5Gi', '2Gi'], {placeHolder: 'Select the storage size'});
-        if (!storageSize) return null;
+    static async getProjectName() {
```

Comment:
Please see https://github.com/redhat-developer/vscode-openshift-tools/pull/432#discussion_r245595571 and #499 for the summary of my issues with this implementation.